### PR TITLE
More improvements on signals handling

### PR DESCRIPTION
**Pull request by s0undt3ch**

### CODE REVIEW

File: salt/utils/reactor.py [hunk 1]
```diff
@@ -213,10 +202,6 @@ def run(self):
         '''
         Enter into the server loop
         '''
-        # Properly exit if a SIGTERM/SIGINT is signalled
```
Comment (cachedout) [inline salt/utils/reactor.py]: Please remove the import since this is now unused.

Comment (s0undt3ch) [inline salt/utils/reactor.py]: Will do.

**Comment by s0undt3ch** [comment]
Are the failures also happening on current develop?

**Comment by cachedout** [comment]
Unfortunately, we have no testing on develop right now. All PRs are on hold until it is fixed.

**Comment by s0undt3ch** [comment]
Bummer.... OK.

Removed the import.

File: salt/transport/zeromq.py [hunk 2]
```diff
@@ -728,6 +737,8 @@ def publish(self, load):
             int_payload['topic_lst'] = load['tgt']
 
         pub_sock.send(self.serial.dumps(int_payload))
+        pub_sock.close()
```
Comment (s0undt3ch) [inline salt/transport/zeromq.py]: Maybe set a linger to 5? ie, wait 5 seconds before dropping anything being sent an close?

**Comment by s0undt3ch** [comment]
On some occasions we'd have to send the master a SIGTERM twice in order for it to shutdown. This PR addresses that.

I went a little further and tried to see where ZMQ sockets and context weren't being closed and added code to explicitly close them.

**Comment by cachedout** [comment]
Looks good! Thanks, @s0undt3ch

**Pull request merged**